### PR TITLE
/dev/crypto engine: add missing RC4 parameter

### LINKED ISSUE #7280: devcrypto RC4 table init issue
In the devcrypto code you have the following cipher definition table where the RC4 is missing a 'mode' value and when requesting the algorithm from /dev/crypto uses an ID of 0 which is an invalid algorithm.

```
    { NID_aes_128_cbc, 16, 128 / 8, 16, EVP_CIPH_CBC_MODE, CRYPTO_AES_CBC },
    { NID_aes_192_cbc, 16, 192 / 8, 16, EVP_CIPH_CBC_MODE, CRYPTO_AES_CBC },
    { NID_aes_256_cbc, 16, 256 / 8, 16, EVP_CIPH_CBC_MODE, CRYPTO_AES_CBC },
#ifndef OPENSSL_NO_RC4
    { NID_rc4, 1, 16, 0, CRYPTO_ARC4 },
#endif
```

### PR DESCRIPTION
**Pull request by levitte**
Fixes #7280

### CODE REVIEW

**Comment by levitte** [comment]
Issue submitter confirmed that this PR does its job, see https://github.com/openssl/openssl/issues/7280#issuecomment-423195381

**Review by mattcaswell — APPROVED** [review]

File: crypto/engine/eng_devcrypto.c [hunk 1]
```diff
@@ -69,7 +69,7 @@ static const struct cipher_data_st {
     { NID_aes_192_cbc, 16, 192 / 8, 16, EVP_CIPH_CBC_MODE, CRYPTO_AES_CBC },
     { NID_aes_256_cbc, 16, 256 / 8, 16, EVP_CIPH_CBC_MODE, CRYPTO_AES_CBC },
 #ifndef OPENSSL_NO_RC4
-    { NID_rc4, 1, 16, 0, CRYPTO_ARC4 },
+    { NID_rc4, 1, 16, 0, 0, CRYPTO_ARC4 },
```
[the remark this answers is not in the document]
Comment (levitte) [reply crypto/engine/eng_devcrypto.c]: Oooh!

[the remark this answers is not in the document]
Comment (levitte) [reply crypto/engine/eng_devcrypto.c]: Okie, changed

**Comment by levitte** [comment]
Because of small change, this needs reapproving

**Review by mspncp — APPROVED** [review]

**Comment by levitte** [comment]
Thank you
Merged

master:
f52f2c1ae8f2ec378c5a20fb4104aa38edbabfcb /dev/crypto engine: add missing RC4 parameter

1.1.1:
baa5cdad88590188d5547347e7fc544af40ec8ce /dev/crypto engine: add missing RC4 parameter

**Pull request closed without merging**